File: typescript/src/features/goose/session_manager.ts
// Re-implementation of Goose's Session and Memory Management

import { Message } from './provider_interface';

export class Session {
  public id: string;
  public history: Message[] = [];

  constructor(id: string) {
    this.id = id;
  }

  public addMessage(message: Message) {
    this.history.push(message);
  }

  public getHistory(): Message[] {
    return this.history;
  }
}

export class SessionManager {
  private sessions: Map<string, Session> = new Map();

  public createSession(id: string): Session {
    const session = new Session(id);
    this.sessions.set(id, session);
    return session;
  }

  public getSession(id: string): Session | undefined {
    return this.sessions.get(id);
  }

  public clearSession(id: string) {
    this.sessions.delete(id);
  }
}

File: java/src/main/java/com/tormentnexus/goose/SessionManager.java
package com.tormentnexus.goose;

// Re-implementation of Goose's Session and Memory Management

import com.tormentnexus.goose.ProviderInterface.Message;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class SessionManager {

    public static class Session {
        public String id;
        public List<Message> history;

        public Session(String id) {
            this.id = id;
            this.history = new ArrayList<>();
        }

        public void addMessage(Message message) {
            this.history.add(message);
        }

        public List<Message> getHistory() {
            return this.history;
        }
    }

    private final Map<String, Session> sessions = new HashMap<>();

    public Session createSession(String id) {
        Session session = new Session(id);
        sessions.put(id, session);
        return session;
    }

    public Session getSession(String id) {
        return sessions.get(id);
    }

    public void clearSession(String id) {
        sessions.remove(id);
    }
}
